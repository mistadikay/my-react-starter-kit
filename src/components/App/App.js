// @flow

import React from 'react';

import styles from './index.css';

type Props = {
  children?: React$Element<*>
};

export default class extends React.Component {
  props: Props

  render(): React$Element<*> {
    return (
      <div className={ styles.app }>
        { this.props.children }
      </div>
    );
  }
}
