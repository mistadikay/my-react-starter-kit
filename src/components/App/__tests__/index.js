/* eslint-env es6, jest */
import React from 'react';
import { shallow } from 'enzyme';

import App from '../App';

test('App', () => {
  const component = shallow(
    <App/>
  );

  expect(component).toMatchSnapshot();
});
