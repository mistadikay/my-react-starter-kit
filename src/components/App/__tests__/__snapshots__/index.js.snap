// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="app"
/>,
  "nodes": Array [
    <div
      className="app"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <_class />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": _class {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="app"
/>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="app"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <_class />,
}
`;
